internal/pkg/cli/task_run.go

```diff
@@ -221,28 +428,33 @@ Run a task with environment variables.
 			if err := opts.Ask(); err != nil {
 				return err
 			}
+
+			if err := opts.Execute(); err != nil {
+				return err
+			}
 			return nil
 		}),
 	}
 
-	cmd.Flags().Int8Var(&vars.count, countFlag, 1, countFlagDescription)
-	cmd.Flags().Int16Var(&vars.cpu, cpuFlag, 256, cpuFlagDescription)
-	cmd.Flags().Int16Var(&vars.memory, memoryFlag, 512, memoryFlagDescription)
+	cmd.Flags().Int64Var(&vars.count, countFlag, 1, countFlagDescription)
+	cmd.Flags().IntVar(&vars.cpu, cpuFlag, 256, cpuFlagDescription)
+	cmd.Flags().IntVar(&vars.memory, memoryFlag, 512, memoryFlagDescription)
 
 	cmd.Flags().StringVarP(&vars.groupName, taskGroupNameFlag, nameFlagShort, "", taskGroupFlagDescription)
 
 	cmd.Flags().StringVar(&vars.image, imageFlag, "", imageFlagDescription)
 	cmd.Flags().StringVar(&vars.dockerfilePath, dockerFileFlag, "", dockerFileFlagDescription)
+	cmd.Flags().StringVar(&vars.imageTag, imageTagFlag, "", taskImageTagDescription)
 
 	cmd.Flags().StringVar(&vars.taskRole, taskRoleFlag, "", taskRoleFlagDescription)
 
 	cmd.Flags().StringVar(&vars.appName, appFlag, "", appFlagDescription)
 	cmd.Flags().StringVar(&vars.env, envFlag, "", envFlagDescription)
-	cmd.Flags().StringVar(&vars.subnet, subnetFlag, "", subnetFlagDescription)
+	cmd.Flags().StringSliceVar(&vars.subnets, subnetsFlag, nil, subnetsFlagDescription)
 	cmd.Flags().StringSliceVar(&vars.securityGroups, securityGroupsFlag, nil, securityGroupsFlagDescription)
 
 	cmd.Flags().StringToStringVar(&vars.envVars, envVarsFlag, nil, envVarsFlagDescription)
-	cmd.Flags().StringSliceVar(&vars.commands, commandsFlag, nil, commandsFlagDescription)
+	cmd.Flags().StringVar(&vars.command, commandFlag, "", commandFlagDescription)
```

Comment:
Now that I'm thinking about this design and how docker/kubectl do things - wonder if we should take the CMD as an arg, rather than a flag. What do you think about that UX?

Reply:
I think that makes sense - considering that `docker run` accepts it as an optional argument and that the popular use case is to run the same image with different env vars or command.

One concern on top of my head is the mixture of flags and arguments could be confusing sometimes. An argument could be confused as a flag value depending on the position at which the argument is provided. I know I always mess that up as a user😥.

Reply:
In the style guide for the CLI we decided: https://github.com/aws/copilot-cli/blob/master/STYLE_GUIDE.md#flags
> Flags are used when you need more than one required input of different type or if the inputs are optional.

Since the flag is optional and to keep it consistent with the other commands, I think we should have it behind a flag